macros/models/contracts/find_self_destruct_contracts.sql

```diff
@@ -31,14 +31,14 @@ FROM (
           = (CASE WHEN cardinality(sd.trace_address) = 0 then cast(-1 as bigint) else sd.trace_address[1] end)
       and sd.type = 'suicide'
       {% if is_incremental() %}
-      and sd.block_time >= date_trunc('day', now() - interval '7' day)
+      AND {{ incremental_predicate('sd.block_time') }}
       and cr.address NOT IN (SELECT contract_address FROM {{this}} ) --ensure no duplicates
```

Comment:
is this really needed?
`and cr.address NOT IN (SELECT contract_address FROM {{this}} ) --ensure no duplicates`

address is already the unique key on the spell, so the merge statement built will ensure no dupes since it joins on that column and only inserts/updates as needed

i wonder if this extra filter against itself is causing a slowdown

Reply:
Without this, we'd get duplicate contracts in the CTEs, which may screw up some of the joins and ranking columns